content/800-guides/160-tanstack-start.mdx

```diff
@@ -0,0 +1,427 @@
+      },
+      {
+        title: "Prisma Tanstack Start Demo",
+      },
+    ],
+  }),
+  component: RootComponent,
+});
+
+function RootComponent() {
+  return (
+    <RootDocument>
+      <Outlet />
+    </RootDocument>
+  );
+}
+
+function RootDocument({ children }: Readonly<{ children: ReactNode }>) {
+  return (
+    <html>
+      <head>
+        <HeadContent />
+      </head>
+      <body>
+        {children}
+        <Scripts />
+      </body>
+    </html>
+  );
+}
+```
+
+Create one more file called `index.tsx` in `app/routes/`:
+
+```typescript file=app/routes/index.tsx showLineNumbers
+import { createFileRoute } from "@tanstack/react-router";
+
+export const Route = createFileRoute("/")({
+  component: Home,
+});
+
+function Home() {
+  return (
+    <div>
+      <h1>Posts</h1>
+    </div>
+  );
+}
+```
+
+Now, run:
+
+```terminal
+npm run dev
+```
+
+This will generate the `routeTree.gen.ts` file and resolve any routing errors.
+
+Your file tree should look like this:
+
+```
+.
+├── app/
+│   ├── routes/
+│   │   ├── __root.tsx
+│   │   └── index.tsx
+│   ├── client.tsx
+│   ├── router.tsx
+│   ├── routeTree.gen.ts
+│   └── ssr.tsx
+├── .gitignore
+├── app.config.ts
+├── package-lock.json
+├── package.json
+└── tsconfig.json
+```
+
+## Step 2: Install and Configure Prisma ORM
+
+Next, we need to install and set up Prisma ORM in our project. Let's start by installing the Prisma CLI.
+
+### 2.1 Install Prisma ORM and Define Your Model
+
+To start, install the necessary dependencies. At the root of your project, run:
+
+```terminal
+npm install prisma --save-dev
+npm install tsx --save-dev
+npm install @prisma/extension-accelerate
+```
+
+The Prisma CLI is now installed.
+
+Now, run the following command to set up Prisma:
+
+```terminal
+npx prisma init --db
+```
+
+This does a few things:
+
+- Creates a `prisma` directory with a `schema.prisma` file.
+- Creates a Prisma Postgres database.
+- Creates a `.env` file containing the `DATABASE_URL` at the project root.
+
+In `schema.prisma`, create a model for our Posts:
+
+```prisma file=prisma/schema.prisma showLineNumbers
+model Post {
+  id      Int     @id @default(autoincrement())
+  title   String
+  content String?
+}
+```
+
+To migrate your database, run:
+
+```terminal
+npx prisma migrate dev --name init
+```
+
+This does three things:
+
+- Creates a new SQL migration file in the `prisma/migrations` directory.
+- Executes the SQL migration file against the database.
+- Runs `prisma generate` under the hood, which installs the `@prisma/client` package and generates a tailored Prisma Client API based on your models.
+
+To view your database, run:
+
+```terminal
+npx prisma studio
+```
+
+Select the `Post` table and add a record.
+
+<Admonition type="info">
+
+Instead of using Prisma Studio, you can seed your database with data using the following command. See [Prisma Seed](https://www.prisma.io/docs/orm/prisma-migrate/workflows/seeding) for more information.
```

Comment:
```suggestion
Instead of using Prisma Studio, you can seed your database with data using the following command. See [Prisma Seed](/orm/prisma-migrate/workflows/seeding) for more information.
```